sdk/python/kfp/_client.py

```diff
@@ -795,7 +818,16 @@ def create_recurring_run(
 
         Returns:
           A Job object. Most important field is id.
+
+        Raises:
+          ValueError: If required parameters are not supplied.
         """
+        if experiment_id is None and experiment_name is None:
+            raise ValueError('Either job_id or experiment_name is required')
+        if not experiment_id:
+            experiment = self.create_experiment(experiment_name)
```

Comment:
So, `experiment_id` is for retrieving an existing experiment while `experiment_name` is for creating a new experiment. This could be confusing. 
Why not just create the experiment separately and always use `experiment_id` in this method?

Reply:
Yeah true, looks like I got my wires crossed on this one. Really I was looking to replicate the behavior of the `kfp run submit` CLI method which looks like it handles `client.create_experiment` under the hood, the mistake I've made here is pushing it down to the client level, definitely agree `experiment_name` shouldn't be needed here. Let me rectify this, thanks for catching!

Reply:
Also reverted this change too.